src/assets/wise5/authoringTool/peer-grouping/edit-peer-grouping-dialog/edit-peer-grouping-dialog.component.ts

```diff
@@ -32,18 +37,37 @@ export class EditPeerGroupingDialogComponent extends AuthorPeerGroupingDialogCom
     this.logicType = this.getLogicType(this.peerGrouping.logic);
     if (this.logicType === DIFFERENT_IDEAS_VALUE) {
       this.referenceComponent = this.getDifferentIdeasReferenceComponent(this.peerGrouping.logic);
+    } else if (this.logicType === DIFFERENT_SCORES_VALUE) {
+      this.referenceComponent = this.getDifferentScoresReferenceComponent(this.peerGrouping.logic);
+      this.mode = this.getDifferentScoresMode(this.peerGrouping.logic);
     }
   }
 
   private getLogicType(logic: string): string {
-    return new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic) != null ? DIFFERENT_IDEAS_VALUE : logic;
+    if (new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic) != null) {
+      return DIFFERENT_IDEAS_VALUE;
+    } else if (new RegExp(DIFFERENT_SCORES_REGEX).exec(logic) != null) {
+      return DIFFERENT_SCORES_VALUE;
+    } else {
+      return logic;
+    }
   }
 
   private getDifferentIdeasReferenceComponent(logic: string): ReferenceComponent {
     const result = new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic);
     return new ReferenceComponent(result[1], result[2]);
   }
 
+  private getDifferentScoresReferenceComponent(logic: string): ReferenceComponent {
+    const result = new RegExp(DIFFERENT_SCORES_REGEX).exec(logic);
+    return new ReferenceComponent(result[1], result[2]);
+  }
+
+  private getDifferentScoresMode(logic: string): string {
+    const result = new RegExp(DIFFERENT_SCORES_REGEX).exec(logic);
+    return result[4];
+  }
```

Comment:
Can these functions be in the PeerGrouping class instead?

Change the line
```
this.peerGrouping = this.utilService.makeCopyOfJSONObject(this.peerGrouping);
```
to
```
this.peerGrouping = new PeerGrouping(this.utilService.makeCopyOfJSONObject(this.peerGrouping));
```
to make this.peerGrouping an instance of the PeerGrouping class so that we can call those functions?